pkg/kubelet/cm/cpumanager/policy_static.go

```diff
@@ -87,7 +110,14 @@ var _ Policy = &staticPolicy{}
 // NewStaticPolicy returns a CPU manager policy that does not change CPU
 // assignments for exclusively pinned guaranteed containers after the main
 // container process starts.
-func NewStaticPolicy(topology *topology.CPUTopology, numReservedCPUs int, reservedCPUs cpuset.CPUSet, affinity topologymanager.Store) (Policy, error) {
+func NewStaticPolicy(topology *topology.CPUTopology, numReservedCPUs int, reservedCPUs cpuset.CPUSet, affinity topologymanager.Store, cpuPolicyOptions map[string]string) (Policy, error) {
+	opts, err := NewStaticPolicyOptions(cpuPolicyOptions)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
This should be surrounded by the feature flag, right? There's no way to get the previous behaviour back otherwise.

Reply:
We used to have a feature flag check but it was moved earlier in the flow (https://github.com/kubernetes/kubernetes/pull/101432/commits/91e6a8c4fbdd2bbbf607fc53e3fbf1fe705c4ff9). Once we are here, we can't possibly get the options (in the production code) unless we have the right set of feature flags.

Reply:
Agree with @fromanirh on the desired flow. All feature flag checks are done in the `containermanager`, with an empty set of options passed in if none are specified or the feature flag is turned off. This is consistent with how similar feature flags are treated for the topologymanager and devicemanager as well.